src/response_code.rs

```diff
@@ -476,6 +504,15 @@ impl std::fmt::Display for Error {
     }
 }
 
+impl std::error::Error for Error {
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+        match self {
+            Error::WrapperError(wrapper_error) => Some(wrapper_error),
```

Comment:
Same thing here ...
```rust
Error::WrapperError(ref wrapper_error) => Some(wrapper_error),
Error::Tss2Error(ref response_code) => Some(response_code),
```

But again I am not sure it is necessary. Maybe it is implicitly converted to a reference due to the way the return type is declared?

Reply:
I think it's implicit because the function takes in `&self`, so `match self` actually returns `Error::WrapperError(&WrapperErrorKind)`, not an actual object!

Reply:
Yeah makes sense. In that case I have no more questions. Nice job.